numba_dppy/utils/array_utils.py

```diff
@@ -30,6 +30,42 @@
 ]
 
 
+def get_info_from_suai(obj):
+    """
+    Convenience function to gather information from __sycl_usm_array_interface__.
+
+    Args:
+        obj: Array with SUAI attribute.
+
+    Returns:
+        usm_mem: USM memory object.
+        total_size: Total number of items in the array.
+        shape: Shape of the array.
+        ndim: Total number of dimensions.
+        itemsize: Size of each item.
+        strides: Stride of the array.
+        dtype: Dtype of the array.
+    """
+    usm_mem = dpctl_mem.as_usm_memory(obj)
+
+    assert usm_mem is not None
+
+    shape = obj.__sycl_usm_array_interface__["shape"]
+    total_size = np.prod(obj.__sycl_usm_array_interface__["shape"])
+    ndim = len(obj.__sycl_usm_array_interface__["shape"])
+    itemsize = np.dtype(obj.__sycl_usm_array_interface__["typestr"]).itemsize
+    dtype = np.dtype(obj.__sycl_usm_array_interface__["typestr"])
+    strides = obj.__sycl_usm_array_interface__["strides"]
+    if strides is None:
+        strides = [
+            1,
+        ] * ndim
+        for i in reversed(range(1, ndim)):
+            strides[i - 1] = strides[i] * shape[i]
+        strides = tuple(strides)
+    return usm_mem, total_size, shape, ndim, itemsize, strides, dtype
```

Comment:
Instead of returning many values, it is better to return structure or adaptor class.